### PR TITLE
Remote Wallet: Stricter derivation path component parsing

### LINKED ISSUE #10722: Ledger app returns wallet ID when presented with negative key path
#### Problem
Passing a negative value to the solana ledger app will return the Ledger's wallet ID.
```
~$ solana --version
solana-cli 1.3.0 296ac10b
~$ solana-keygen pubkey usb://ledger
2JT2Xvy6T8hSmT8g6WdeDbHUgoeGdj6bE2VueCZUJmyN
~$ solana-keygen pubkey usb://ledger?key=-1
2JT2Xvy6T8hSmT8g6WdeDbHUgoeGdj6bE2VueCZUJmyN
```
To my knowledge, a negative value here is an invalid derivation path.

#### Proposed Solution
Return an error message when invalid key paths are passed.

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Lax derivation path component parsing in remote wallet

#### Summary of Changes

Introduce `DerivationPathComponent` type
Surface parse errors

Fixes #10722 
Closes #10723 

cc/ @danpaul000

### CODE REVIEW

File: remote-wallet/src/remote_wallet.rs [hunk 1]
```diff
@@ -322,20 +324,61 @@ impl RemoteWalletInfo {
...
+            ))
+        })
+    }
+}
+
+impl std::fmt::Display for DerivationPathComponent {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(fmt, "{}", self.0)
+    }
+}
```
Comment (CriesofCarrots) [inline remote-wallet/src/remote_wallet.rs:366]: Can we rework this? Right now, if I debug print the DerivationPath after parsing, it looks like this for `?key=1`, which is both confusing and actually incorrect (due to the apostrophe) 😞 :
```
m/44'/501'/DerivationPathComponent(2147483649)'
```

Would prefer this output for at least Display, and maybe Debug as well:
```
m/44'/501'/1'
```

You'll probably want to move the addition of the apostrophes into Display/Debug for DerivationPathComponent, currently here:
https://github.com/solana-labs/solana/blob/d42247c6524f249247ab39eb28cfe0da3069362e/remote-wallet/src/remote_wallet.rs#L333

Comment (t-nelson) [reply remote-wallet/src/remote_wallet.rs]: Updated! https://github.com/solana-labs/solana/pull/10725/commits/493c19a5fbfa6b12f260c9d1a1eef37514927f78#diff-7ef79c2848cdef4bdf705e9d9000c5dfR660-R662

File: remote-wallet/src/remote_wallet.rs [hunk 2]
```diff
@@ -322,20 +324,73 @@ impl RemoteWalletInfo {
...
-    pub change: Option<u32>,
+    pub account: Option<DerivationPathComponent>,
+    pub change: Option<DerivationPathComponent>,
 }
 
 impl fmt::Debug for DerivationPath {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let account = if let Some(account) = self.account {
+        let account = if let Some(account) = &self.account {
             format!("/{:?}'", account)
```
Comment (CriesofCarrots) [inline remote-wallet/src/remote_wallet.rs:389]: Can you please remove the apostrophe here and on 394?

Comment (t-nelson) [reply remote-wallet/src/remote_wallet.rs]: Done!  Added a test too.  https://github.com/solana-labs/solana/pull/10725/commits/2e6c29ff6c1c343b47244f533a86bb02140efe1d#diff-7ef79c2848cdef4bdf705e9d9000c5dfR648-R658

**Pull request merged**